tests/zfs-tests/tests/functional/cli_root/zfs_set/zfs_set_keylocation.ksh

```diff
@@ -64,7 +64,13 @@ log_must zfs create -o encryption=on -o keyformat=passphrase \
 	-o keylocation=file:///$TESTPOOL/pkey $TESTPOOL/$TESTFS1
 
 log_mustnot zfs set keylocation=none $TESTPOOL/$TESTFS1
-log_mustnot zfs set keylocation=/$TESTPOOL/pkey $TESTPOOL/$TESTFS1
+if is_linux; then
+	log_mustnot zfs set keylocation=/$TESTPOOL/pkey $TESTPOOL/$TESTFS1
+else
+	# file:///$TESTPOOL/pkey and /$TESTPOOL/pkey are equivalent on FreeBSD
+	# thanks to libfetch
+	log_must zfs set keylocation=/$TESTPOOL/pkey $TESTPOOL/$TESTFS1
```

Comment:
We should update the comment to make it clear that we want this to work on all platforms either via libfetch or some other library.